tools/vast-regenerate/main.cpp

```diff
@@ -0,0 +1,306 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/concept/parseable/vast/uuid.hpp"
+
+#include <vast/concept/printable/to_string.hpp>
+#include <vast/concept/printable/vast/uuid.hpp>
+#include <vast/detail/filter_dir.hpp>
+#include <vast/detail/logger_formatters.hpp>
+#include <vast/fbs/utils.hpp>
+#include <vast/index_statistics.hpp>
+#include <vast/io/read.hpp>
+#include <vast/io/write.hpp>
+#include <vast/partition_synopsis.hpp>
+#include <vast/synopsis_factory.hpp>
+#include <vast/system/configuration.hpp>
+#include <vast/system/index.hpp>
+#include <vast/uuid.hpp>
+#include <vast/value_index_factory.hpp>
+
+#include <fmt/core.h>
+
+#include <filesystem>
+
+int regenerate_mdx(const std::filesystem::path& dbdir) {
```

Comment:
With how complex this functionality is, should we maybe turn it into a command plugin with its own unit- and integration-tests?

I think the same also applies for `lsvast`, which has integration tests defined in a very complicated way, because it predates the plugin API.

Reply:
Not sure if that makes sense with the current plugin API at least, since both of these tools are supposed to run without a VAST node, and afaik that's currently not possible for a plugin.

Reply:
We can probably add an option to the `command_plugin` scaffolding that allows them to run exclusively, i.e., lock the PID file.